topoembedx/classes/hope.py

```diff
@@ -0,0 +1,174 @@
+"""Higher Order Laplacian Positional Encoder (HOPE) class."""
+import networkx as nx
+import numpy as np
+from scipy import sparse
+
+from topoembedx.neighborhood import neighborhood_from_complex
+
+
+class HOPE:
+    """Higher Order Laplacian Positional Encoder (HOPE) class.
+
+    Parameters
+    ----------
+    dimensions : int, optional
+        Dimensionality of embedding. Defaults to 3.
+    """
+
+    def __init__(self, dimensions: int = 2):
+
+        self.A = []
+        self.ind = []
+        self.dimensions = dimensions
+
+        self._embedding = []
+
+    @staticmethod
+    def _laplacian_pe(A, k: int, return_eigenval: bool = False):
+        """
+        Compute Laplacian Positional Encodings (PE) for a given adjacency matrix.
```

Comment:
Docstring need to be next to """. See our guidelines about docstrings on our doc website.